File: app/src/main/java/org/fossasia/openevent/utils/CommonTaskLoop.java
package org.fossasia.openevent.utils;

/**
 * User: MananWason
 * Date: 8/3/2015
 */
public class CommonTaskLoop {
    private static CommonTaskLoop ourInstance = new CommonTaskLoop();

    private CommonEventLoop commonEventLoop;

    private CommonTaskLoop() {
        commonEventLoop = new CommonEventLoop();
    }

    public static CommonTaskLoop getInstance() {
        return ourInstance;
    }

    public void post(Runnable call) {
        commonEventLoop.post(call);
    }

    public void delayPost(Runnable call, int nMillSec) {
        commonEventLoop.delayPost(call, nMillSec);
    }

    public void shutdown() {
        commonEventLoop.shutdown();
    }
}

File: android/app/src/main/java/org/fossasia/openevent/common/utils/CommonTaskLoop.kt
package org.fossasia.openevent.common.utils

class CommonTaskLoop private constructor() {

    private val commonEventLoop: CommonEventLoop = CommonEventLoop()

    fun post(call: Runnable) {
        commonEventLoop.post(call)
    }

    fun delayPost(call: Runnable, nMillSec: Int) {
        commonEventLoop.delayPost(call, nMillSec)
    }

    fun shutdown() {
        commonEventLoop.shutdown()
    }

    companion object {
        @JvmStatic
        val instance = CommonTaskLoop()
    }
}
